presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -614,6 +614,7 @@ public ListeningExecutorService createResourceManagerExecutor(ResourceManagerCon
         // server info resource
         jaxrsBinder(binder).bind(ServerInfoResource.class);
         jsonCodecBinder(binder).bindJsonCodec(ServerInfo.class);
+        thriftCodecBinder(binder).bindThriftCodec(ServerInfo.class);
```

Comment:
Is this required for your tests to pass? IIRC this is  only required for actual wiring during presto server startup.